keyvi/include/keyvi/dictionary/fsa/internal/minimization_hash.h

```diff
@@ -82,19 +83,28 @@ class MinimizationHash final {
     }
   }
 
-  MemoryLimitConfiguration FindMemoryLimitConfiguration(const size_t memory_limit, const size_t min,
-                                                        const size_t max) const {
+  MemoryLimitConfiguration FindMemoryLimitConfigurationForLRUCache(const size_t memory_limit, const size_t min,
+                                                                   const size_t max) const {
     size_t best_fit_memory_limit = 0;
     size_t best_fit_generations = 0;
     size_t best_fit_maximum_number_of_items_per_table = 0;
 
+    const size_t min_memory_limit =
+        (hash_size_step_table_[hash_min_size_step_] + (hash_size_step_table_[hash_min_size_step_] >> 2)) * sizeof(T) *
+        min;
+
+    if (min_memory_limit > memory_limit) {
+      throw std::invalid_argument("memory limit too low, for the given parameters the limit must be at least " +
+                                  std::to_string(min_memory_limit));
```

Comment:
This exception message doesn't say is the minimum in bytes, kilo-bytes or other unit.
I believe it will be better to be explicit here, so users can make a better use of this error message.